io/io/test/TFileMergerTests.cxx

```diff
@@ -0,0 +1,52 @@
+#include "TFileMerger.h"
+
+#include "TMemFile.h"
+#include "TTree.h"
+
+#include "gtest/gtest.h"
+
+static void CreateATuple(TMemFile &file, const char *name, double value)
+{
+   auto mytree = new TTree(name, "A tree");
+   mytree->SetDirectory(&file);
+   mytree->Branch(name, &value);
+   mytree->Fill();
+   file.Write();
+}
+
+static void CheckTree(TMemFile &file, const char *name, double expectedValue)
+{
+   auto t = static_cast<TTree *>(file.Get(name));
+   ASSERT_TRUE(t != nullptr);
+
+   double d;
+   t->SetBranchAddress(name, &d);
+   t->GetEntry(0);
+   EXPECT_EQ(expectedValue, d);
+   // Setting branch address to a stack address requires to either call ResetBranchAddresses or delete the tree.
+   t->ResetBranchAddresses();
+}
+
+TEST(TFileMerger, CreateWithTFilePointer)
+{
+   TMemFile a("a.root", "CREATE");
+   CreateATuple(a, "a_tree", 1.);
+
+   // FIXME: Calling this out of order causes two values to be written to the second file.
+   TMemFile b("b.root", "CREATE");
+   CreateATuple(b, "b_tree", 2.);
+
+   TFileMerger merger;
+   auto output = std::unique_ptr<TMemFile>(new TMemFile("output.root", "CREATE"));
+   output->ResetBit(kMustCleanup);
```

Comment:
This should be done inside the `TFileMerger` to avoid crashes in user code. I am in general quite uneasy about mixing up ROOT managed objects and smart pointers, because that's very likely to lead to problems. However, the benefit of being able to use a TMemFile output may be worth it if we are very careful to avoid exposing these things to the users.

Reply:
This call is redundant from what I see. My local version has it removed.

Reply:
Also, note that `TFileMerger` itself calls [`fOutputFile->SetBit(kMustCleanup);`](https://github.com/root-project/root/blob/master/io/io/src/TFileMerger.cxx#L835) so this `ResetBit()` does nothing, really.